starknet_py/net/schemas/rpc.py

```diff
@@ -510,3 +522,171 @@ class PendingTransactionsSchema(Schema):
     @post_load
     def make_dataclass(self, data, **kwargs):
         return data["pending_transactions"]
+
+
+# ------------------------------- Trace API -------------------------------
+
+
+class EventContentSchema(Schema):
+    keys = fields.List(Felt(), data_key="keys", required=True)
+    data = fields.List(Felt(), data_key="data", required=True)
+
+    @post_load
+    def make_dataclass(self, data, **kwargs):
+        return EventContent(**data)
+
+
+class FunctionInvocationSchema(Schema):
+    contract_address = Felt(data_key="contract_address", required=True)
+    entry_point_selector = Felt(data_key="entry_point_selector", required=True)
+    calldata = fields.List(Felt(), data_key="calldata", required=True)
+    caller_address = Felt(data_key="caller_address", required=True)
+    class_hash = Felt(data_key="class_hash", required=True)
+    entry_point_type = EntryPointTypeField(data_key="entry_point_type", required=True)
+    call_type = CallTypeField(data_key="call_type", required=True)
+    result = fields.List(Felt(), data_key="result", required=True)
+    # https://marshmallow.readthedocs.io/en/stable/nesting.html#nesting-a-schema-within-itself
+    calls = fields.List(
+        fields.Nested(
+            lambda: FunctionInvocationSchema()  # pylint: disable=unnecessary-lambda
+        ),
+        data_key="calls",
+        required=True,
+    )
+    events = fields.List(
+        fields.Nested(EventContentSchema()), data_key="events", required=True
+    )
+    messages = fields.List(
+        fields.Nested(L2toL1MessageSchema()), data_key="messages", required=True
+    )
```

Comment:
Are all of these required?

Reply:
In spec those are specified as [`"allOf"`](https://github.com/starkware-libs/starknet-specs/blob/v0.5.0/api/starknet_trace_api_openrpc.json#L290)

Reply:
Does that imply that all of them are required?

Reply:
Hard to say, sometimes `required` is specified inside `allOf`, sometimes not. It's not specified in v0.5.0 either, I'll ask a question under the PR regarding that

Reply:
Not all of them, mentioned here https://github.com/software-mansion/starknet.py/pull/1175/files/089c91c2a334930384bc2cffc11d238c92e2e51a#r1320093740